gatherers/gather.js

```diff
@@ -16,9 +16,24 @@
  */
 'use strict';
 
+function pendingPromiseFactory() {
```

Comment:
Oh I get it. It's to set up a Promise that can be resolved when the SW data arrives